src/fileio/impls/hdf5/h5_format.cxx

```diff
@@ -348,21 +349,21 @@ bool H5Format::addVar(const std::string &name, bool repeat, hid_t write_hdf5_typ
       // Negative value indicates error, i.e. file does not exist, so create:
       hsize_t init_size[3];
       if (parallel) {
-        init_size[0] = mesh->GlobalNx - 2 * mesh->xstart;
+        init_size[0] = lx == 0 ? mesh->GlobalNx - 2 * mesh->xstart : lx;
         if (datatype == "FieldPerp") {
-          init_size[1] = mesh->GlobalNy - mesh->numberOfYBoundaries() * 2 * mesh->ystart;
+          init_size[1] = lz == 0 ? mesh->GlobalNz : lz;
         } else {
-          init_size[1] = mesh->GlobalNz;
+          init_size[1] = ly == 0 ? mesh->GlobalNy - mesh->numberOfYBoundaries() * 2 * mesh->ystart : ly;
```

Comment:
I guess you just caught this bug! Do we now have a test that covers it?

Reply:
No test - it's in the `parallel=true` branch which doesn't work... So it's useless, but thought I might as well fix while I spotted it!

Reply:
Fair enough!